service/awsconfig/v10/ebs/ebs.go

```diff
@@ -86,24 +86,41 @@ func (e *EBS) DeleteVolume(ctx context.Context, volumeID string) error {
 
 // DetachVolume detaches an EBS volume. If force is specified data loss may occur. If shutdown is
 // specified the instance will be stopped first.
-func (e *EBS) DetachVolume(ctx context.Context, volumeID string, attachment VolumeAttachment, force bool, shutdown bool) error {
-	e.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("detaching EBS volume %s from instance %s", volumeID, attachment.InstanceID))
-
+func (e *EBS) DetachVolume(ctx context.Context, volumeID string, attachment VolumeAttachment, force bool, shutdown bool, wait bool) error {
 	if shutdown {
 		e.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("stopping instance %s", attachment.InstanceID))
 
-		_, err := e.client.StopInstances(&ec2.StopInstancesInput{
+		i := &ec2.StopInstancesInput{
 			InstanceIds: []*string{
 				aws.String(attachment.InstanceID),
 			},
-		})
+		}
+		_, err := e.client.StopInstances(i)
 		if err != nil {
 			return microerror.Mask(err)
 		}
 
+		if wait {
+			e.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("waiting for instance %s being stopped", attachment.InstanceID))
+
+			i := &ec2.DescribeInstancesInput{
+				InstanceIds: []*string{
+					aws.String(attachment.InstanceID),
+				},
+			}
+			err := e.client.WaitUntilInstanceStopped(i)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			e.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("waited for instance %s being stopped", attachment.InstanceID))
```

Comment:
`instance has been stopped`? `waited for instance being stopped` cuts my ear.

Reply:
Grammar and expression should be right. Reason for the structure is the other logs. We try to align the log message structure to make scanning them easier for humans.